media/wav.ksy

```diff
@@ -0,0 +1,272 @@
+meta:
+  id: wav
+  file-extension: wav
+  title: Microsoft WAVE audio file
+  endian: le
+  license: BSD-3-Clause-Attribution
+  encoding: ASCII
+doc: |
+  The WAVE file format is a subset of Microsoft's RIFF specification for the
+  storage of multimedia files. A RIFF file starts out with a file header
+  followed by a sequence of data chunks. A WAVE file is often just a RIFF
+  file with a single "WAVE" chunk which consists of two sub-chunks --
+  a "fmt " chunk specifying the data format and a "data" chunk containing
+  the actual sample data.
+  This Kaitai implementation was written by John Byrd of Gigantic Software
+  (jbyrd@giganticsoftware.com), and it is likely to contain bugs.
+doc-ref: http://soundfile.sapp.org/doc/WaveFormat/
+doc-ref: https://www.loc.gov/preservation/digital/formats/fdd/fdd000001.shtml
+seq:
+  - id: riff_id
+    contents: RIFF
+  - id: file_size
+    type: u4
+  - id: wave_id
+    contents: WAVE
+  - id: chunks
+    type: chunks_type
+    size: file_size - 5
+  - id: unused
+    size-eos: true
+    
+instances: 
+  format_chunk:
+    value: _root.chunks.chunk[0].data 
+
+types:  
+  chunks_type:
+    seq:   
+      - id: chunk
+        type: chunk_type
+        repeat: eos
+ 
+  chunk_type:
+    seq: 
+      - id: chunk_id
+        type: u4be
+      - id: len
+        type: u4 
+      - id: data
+        size: len
+        type: 
+          switch-on: chunk_id 
+          cases:
+            0x666d7420: format_chunk_type
+            0x62657874: bext_chunk_type
+            0x63756520: cue_chunk_type
+            0x64617461: data_chunk_type
```

Comment:
This would look better with enums, like

```yaml
cases:
  'chunk_type::fmt': format_chunk_type
  'chunk_type::bext': bext_chunk_type
```

etc.